packages/theme-product/theme-product.js

```diff
@@ -0,0 +1,127 @@
+/**
+ * Search through the product object and return a variant
+ *
+ * @param {Object} product Product JSON object
+ * @param {*} value The targeted value. It accepts :
+ * - Strings/Number (e.g. 520670707773)
+ * - Object with ID key (e.g. { id: 6908198649917 })
+ * - Object with 'name' and 'value' keys (e.g. [{ name: "Size", value: "36" }, { name: "Color", value: "Black" }])
+ * - Array of values: (e.g. ["38", "Black"])
+ * @returns {Object} The variant object.
+ */
+export function getVariant(product, value) {
```

Comment:
I'm wondering if we should split this into multiple public functions -- `getVariantFromId`, `getVariantFromOptionCollection`... is this one function to rule them all good?

Reply:
I like to separate into multiple functions (personally) even though this all-in-one function does everything. I can adjust the test cases accordingly.

Let me know if you want me to proceed ahead.

Reply:
Lets separate them.